nsv13/code/modules/overmap/boarding/boarding.dm

```diff
@@ -1,18 +1,56 @@
+#define DROP_TROOPER_TEAMS list("Noble", "Helljumper","Red", "Black", "Crimson", "Osiris", "Apex", "Apollo", "Thrace", "Galactica", "Valkyrie", "Recon", "Gamma", "Alpha", "Bravo", "Charlie", "Delta", "Indigo", "Sol's fist", "Abassi", "Cartesia", "Switchback", "Majestic", "Mountain", "Shadow", "Shrike", "Sterling", "FTL", "Belter", "Moya", "Crichton")
 
-/obj/structure/overmap/proc/spawn_boarders(amount, faction_selection="syndicate")
+/** Attempts to spawn knpc or player boarders for this ship.
+ * Arguments:
+ * * amount: Number of boarders to spawn. Will be automatically choosen, based on SSovermap mode difficulty if argument not passed in.
+ * * faction_selection: Faction the boarders belong to. Valid values are currently "syndicate", "pirate", "greytide", "nanotrasen"
+ * * min_players: Number of active players below which boarders won't spawn.
+ * * min_players_for_ghosts: Number of active players below which gosts won't be offered to become boarders.
+ * Returns:
+ * * TRUE when boarders were successfully spawned, otherwise FALSE.
+*/
+/obj/structure/overmap/proc/spawn_boarders(amount, faction_selection="syndicate", min_players = 5, min_players_for_ghosts = 20)
+	if(SSovermap_mode.override_ghost_boarders)
+		message_admins("Failed to spawn boarders for [name] due to admin boarding override.")
+		return FALSE //Allows the admins to disable boarders for event rounds
 	if(!length(occupying_levels))
 		message_admins("Failed to spawn boarders for [name], it doesn't seem to have any occupying z-levels. (Interior)")
 		return FALSE
 	var/player_check = get_active_player_count(alive_check = TRUE, afk_check = TRUE, human_check = TRUE)
+	if(player_check < min_players)
+		message_admins("KNPC boarder spawning aborted due to insufficient playercounts.")
+		return FALSE //No... just no. I'm not that mean
+
 	if(!amount)
 		amount = CEILING(1 + (SSovermap_mode.mode.difficulty / 2), 1)
 	var/list/zs = list()
 	for(var/datum/space_level/SL in occupying_levels)
 		zs += SL.z_value
-	var/startside = pick(GLOB.cardinals)
+	var/turf/target = choose_boarding_location(zs)
+	if(!target)
+		message_admins("Failed to spawn boarders for [name], does it have an interior?")
+		return FALSE //Cut off here to avoid polling people for a spawn that will never work.
+	var/list/candidates = list()
 
-	///Finds a "safe" place to dump a boarding pod, with a bit of distance from the transition edge to avoid visual hiccups.
+	//20 or more players? You're allowed "real" boarders.
+	if(player_check >= min_players_for_ghosts) // Remove the low pop boarder camping
+		candidates = pollCandidatesForMob("Do you want to play as a boarding team member?", ROLE_OPERATIVE, null, ROLE_OPERATIVE, 10 SECONDS, src)
+	//No candidates? Well! Guess you get to deal with some KNPCs :))))))
+	if(!LAZYLEN(candidates))
```

Comment:
```suggestion
	if(!length(candidates))
```
lazylist defines should only be used for lazy lists

Reply:
fixed (tho I deflect blame for that to IndusRobot)